apps/dev-api/domains/user/utils/testSchema/index.js

```diff
@@ -0,0 +1,214 @@
+const DEFAULT_TEST_ADMIN_SECRET = conf.DEFAULT_TEST_ADMIN_SECRET || '3a74b3f07978'
+
+function createTestPhone () {
+    const { country_code, mobile_begin_with, phone_number_lengths } = faker.helpers.arrayElement(countryPhoneData.filter(x => get(x, 'mobile_begin_with.length', 0) > 0))
+    const length = max(phone_number_lengths)
+    const code = String(faker.helpers.arrayElement(mobile_begin_with))
+
+    return faker.phone.number('+' + country_code + code + repeat('#', length - code.length))
+}
+
+async function createTestUser (client, extraAttrs = {}) {
+    if (!client) throw new Error('no client')
+    const sender = { dv: 1, fingerprint: faker.random.alphaNumeric(8) }
+
+    const attrs = {
+        dv: 1,
+        sender,
+        name: faker.internet.userName(),
+        phone: createTestPhone(),
+        password: faker.internet.password(),
+        ...extraAttrs,
+    }
+    const obj = await User.create(client, attrs)
+    return [obj, attrs]
+}
+
+async function updateTestUser (client, id, extraAttrs = {}) {
+    if (!client) throw new Error('no client')
+    if (!id) throw new Error('no id')
+    const sender = { dv: 1, fingerprint: faker.random.alphaNumeric(8) }
+
+    const attrs = {
+        dv: 1,
+        sender,
+        ...extraAttrs,
+    }
+    const obj = await User.update(client, id, attrs)
+    return [obj, attrs]
+}
+
+async function createTestConfirmPhoneAction(client, extraAttrs = {}) {
+    if (!client) throw new Error('no client')
+    const sender = { dv: 1, fingerprint: faker.random.alphaNumeric(8) }
+
+    const attrs = {
+        dv: 1,
+        sender,
+        phone: createTestPhone(),
+        code: faker.random.numeric(CONFIRM_ACTION_CODE_LENGTH),
+        expiresAt: dayjs().add(CONFIRM_ACTION_TTL_IN_SEC, 's'),
+        ...extraAttrs,
+    }
+    const obj = await ConfirmPhoneAction.create(client, attrs)
+    return [obj, attrs]
+}
+
+async function updateTestConfirmPhoneAction (client, id, extraAttrs = {}) {
+    if (!client) throw new Error('no client')
+    if (!id) throw new Error('no id')
+    const sender = { dv: 1, fingerprint: faker.random.alphaNumeric(8) }
+
+    const attrs = {
+        dv: 1,
+        sender,
+        ...extraAttrs,
+    }
+    const obj = await ConfirmPhoneAction.update(client, id, attrs)
+    return [obj, attrs]
+}
+
+async function authenticateUserWithPhoneAndPasswordByTestClient(client, attrs = {}) {
+    if (!client) throw new Error('no client')
+
+    const { data, errors } = await client.mutate(AUTHENTICATE_USER_WITH_PHONE_AND_PASSWORD_MUTATION, { data: attrs })
+    throwIfError(data, errors)
+    return [data.result, attrs]
+}
+/* AUTOGENERATE MARKER <FACTORY> */
+
+async function startConfirmPhoneActionByTestClient(attrs = {}, client) {
+    client ??= await makeClient()
+    attrs.dv ??= 1
+    attrs.sender ??= { dv: 1, fingerprint: faker.random.alphaNumeric(8) }
+    attrs.phone ??= createTestPhone()
```

Comment:
What is the reason for using conditional assignment in this case?

Reply:
It's just a bit different syntax from old one. (Instead of):
```
function createTestSomething(extraAttrs = {}) {
    // generate a lot of data
    const attr = {
        // a lot of duplicated data
        ...extraAttrs
    }
}
```